### PR TITLE
Pull latest changes from nodejs/node

### PR DESCRIPTION
**Pull request by aduh95**
Tests are failing on v16.x and v14.x, not sure why.

### CODE REVIEW

**Comment by MoLow** [comment]
@aduh95 any change you open this pr from a branch on this repo so I can check it out and tackle with it?
node 14 is missing an AbortSignal polyfill
and node 16 seems to have some other issue I want to take a look at

**Comment by aduh95** [comment]
> node 14 is missing an AbortSignal polyfill

That's not exact, Node.js v14.x has `AbortSignal` but you need to have `NODE_OPTIONS=--experimental-abortcontroller` in your env.

> any change you open this pr from a branch on this repo so I can check it out and tackle with it?

Any reason why you can't do from my fork? As a maintainer of this repo, you even should be allowed to push commit on my branch if needed.

File: test/message.js
Comment (aduh95) [inline test/message.js:91]: This is a bit unfortunate, I can't find an easy solution for those. If anyone knows how to make it work, please comment.

Comment (MoLow) [reply test/message.js:91]: I think this is due to different implementation of `AbortSignal.abort()` in node versions

```js
Welcome to Node.js v16.15.1.
Type ".help" for more information.
> AbortSignal.abort().reason
```

```js
Welcome to Node.js v18.6.0.
Type ".help" for more information.
> AbortSignal.abort().reason
DOMException [AbortError]: This operation was aborted
    at new DOMException (node:internal/per_context/domexception:53:5)
    at AbortSignal.abort (node:internal/abort_controller:161:14)
    at REPL13:1:13
    at Script.runInThisContext (node:vm:129:12)
    at REPLServer.defaultEval (node:repl:572:29)
    at bound (node:domain:426:15)
    at REPLServer.runBound [as eval] (node:domain:437:12)
    at REPLServer.onLine (node:repl:902:10)
    at REPLServer.emit (node:events:525:35)
    at REPLServer.emit (node:domain:482:12)
```

Comment (aduh95) [reply test/message.js]: Nice work! Thanks :)

**Review by MoLow — APPROVED** [review]

**Review by MoLow — APPROVED** [review]
LGTM, when do we merge such prs?

File: test/common/index.js
Comment (juliangruber) [inline test/common/index.js:105]: will this be required in consumer code too? Would it be safer to use a AbortSignal ponyfill?

Comment (MoLow) [reply test/common/index.js:105]: no, it is not needed, these adjustments are just so the tap output will match across node versions

Comment (juliangruber) [reply test/common/index.js:105]: Great, thank you!

```suggestion
// Ensure test output is consistent across node versions
if (typeof AbortSignal.timeout !== 'function') {
```

Comment (aduh95) [reply test/common/index.js:105]: `AbortSignal.timeout` doesn't exist on Node.js 14.x, so all the tests that use it fails on that version. It's not required though, `test/node-core-test.js` makes sure that the library still works when it's not polyfilled.

The section below, that overrides `AbortSignal.abort` and `AbortController.prototype.abort` is indeed there just for the TAP output to match, and same, we have `test/node-core-test.js` to ensure it's not necessary for the lib to work.

Comment (aduh95) [reply test/common/index.js:105]: I'm going to land this and open a follow-up PR that adds the comments.

**Pull request merged**
